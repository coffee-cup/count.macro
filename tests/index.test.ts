import pluginTester from "babel-plugin-tester";
import * as plugin from "babel-plugin-macros";

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: {
    filename: __filename,
  },
  tests: {
    "no usage": `import { lines, linesIn, words, wordsIn } from "../lib/index.macro"`,
    lines: `
import { lines } from "../lib/index.macro"
const x = lines;
`,
    linesIn: `
import { linesIn } from "../lib/index.macro"
const x = linesIn("./lines.txt");
`,
    words: `
import { words } from "../lib/index.macro"
const x = words;
`,
    wordsIn: `
import { wordsIn } from "../lib/index.macro"
const x = wordsIn("./words.txt");
`,
  },
});
