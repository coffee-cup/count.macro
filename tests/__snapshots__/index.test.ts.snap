// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros lines: lines 1`] = `

import { lines } from "../lib/index.macro"
const x = lines;

      ↓ ↓ ↓ ↓ ↓ ↓

const x = 2;


`;

exports[`macros linesIn: linesIn 1`] = `

import { linesIn } from "../lib/index.macro"
const x = linesIn("./lines.txt");

      ↓ ↓ ↓ ↓ ↓ ↓

const x = 7;


`;

exports[`macros no usage: no usage 1`] = `

import { lines, linesIn, words, wordsIn } from "../lib/index.macro"

      ↓ ↓ ↓ ↓ ↓ ↓



`;

exports[`macros words: words 1`] = `

import { words } from "../lib/index.macro"
const x = words;

      ↓ ↓ ↓ ↓ ↓ ↓

const x = 10;


`;

exports[`macros wordsIn: wordsIn 1`] = `

import { wordsIn } from "../lib/index.macro"
const x = wordsIn("./words.txt");

      ↓ ↓ ↓ ↓ ↓ ↓

const x = 3;


`;
